### PR TITLE
REF use soname for libgfortran

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge-admin, please rerender

**Comment by beckermr** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -87,7 +87,7 @@ outputs:
     requirements:
       host:
         # So that pin_compatible has something to work with:
-        - libgfortran-ng {{ ctng_gcc }}.*
+        - libgfortran{{ libgfortran_soname }} {{ ctng_gcc }}.*
```
Comment (isuruf) [inline recipe/meta.yaml:90]: For mutual exclusivity
```suggestion
        - libgfortran{{ libgfortran_soname }} {{ ctng_gcc }}.*
        - libgfortran-ng
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -97,8 +97,7 @@ outputs:
         - sysroot_{{ ctng_target_platform }}
     run_exports:
       strong:
-        # The SONAME changes at major releases.
-        - {{ pin_compatible("libgfortran-ng", min_pin="x") }}
+        - libgfortran{{ libgfortran_soname }} >={{ ctng_gcc }}
```
Comment (isuruf) [inline recipe/meta.yaml:100]: For mutual exclusivity
```suggestion
        - libgfortran{{ libgfortran_soname }} >={{ ctng_gcc }}
        - libgfortran-ng
```

**Comment by beckermr** [comment]
I need to upload the arch and ppc compilers before we merge this.

**Comment by beckermr** [comment]
@conda-forge-admin restart ci

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
Bump here @isuruf

**Review by isuruf — APPROVED** [review]

**Pull request merged**